File: templates/projects/node-standard/skeleton/src/routes/health.ts
/**
 * Health check routes
 */

import type { FastifyPluginAsync } from 'fastify';

interface HealthResponse {
    status: 'ok' | 'degraded' | 'unhealthy';
    timestamp: string;
    uptime: number;
    version: string;
}

export const healthRoutes: FastifyPluginAsync = async (app) => {
    app.get<{ Reply: HealthResponse }>(
        '/health',
        {
            schema: {
                response: {
                    200: {
                        type: 'object',
                        properties: {
                            status: { type: 'string', enum: ['ok', 'degraded', 'unhealthy'] },
                            timestamp: { type: 'string', format: 'date-time' },
                            uptime: { type: 'number' },
                            version: { type: 'string' },
                        },
                    },
                },
            },
        },
        async () => {
            return {
                status: 'ok',
                timestamp: new Date().toISOString(),
                uptime: process.uptime(),
                version: '0.1.0',
            };
        }
    );

    app.get('/ready', async () => {
        // Add readiness checks here (database, cache, etc.)
        return { ready: true };
    });

    app.get('/live', async () => {
        return { alive: true };
    });
};

File: templates/projects/python-fastapi/skeleton/app/routes/health.py
"""Health check routes."""

from datetime import datetime
from typing import Literal

from fastapi import APIRouter
from pydantic import BaseModel


class HealthResponse(BaseModel):
    """Health check response."""

    status: Literal["ok", "degraded", "unhealthy"]
    timestamp: datetime
    version: str


class ReadyResponse(BaseModel):
    """Readiness check response."""

    ready: bool


router = APIRouter()


@router.get("/health", response_model=HealthResponse)
async def health() -> HealthResponse:
    """Health check endpoint."""
    return HealthResponse(
        status="ok",
        timestamp=datetime.now(),
        version="0.1.0",
    )


@router.get("/ready", response_model=ReadyResponse)
async def ready() -> ReadyResponse:
    """Readiness check endpoint."""
    # Add dependency checks here (database, cache, etc.)
    return ReadyResponse(ready=True)


@router.get("/live")
async def live() -> dict:
    """Liveness check endpoint."""
    return {"alive": True}
